src/modules/src/controller_lee_payload.c

```diff
@@ -437,23 +483,45 @@ void controllerLeePayload(controllerLeePayload_t* self, control_t *control, setp
     struct vec plStPos = mkvec(state->payload_pos.x, state->payload_pos.y, state->payload_pos.z);
     struct vec plStVel = mkvec(state->payload_vel.x, state->payload_vel.y, state->payload_vel.z);
 
+    // rotational states of the payload
+    struct quat plquat = mkquat(state->payload_quat.x, state->payload_quat.y, state->payload_quat.z, state->payload_quat.w);
+    struct vec plomega = mkvec(state->payload_omega.x, state->payload_omega.y, state->payload_omega.z);
     float l = vmag(vsub(plStPos, statePos));
 
     // errors
     struct vec plpos_e = vclampnorm(vsub(plPos_d, plStPos), self->Kpos_P_limit);
     struct vec plvel_e = vclampnorm(vsub(plVel_d, plStVel), self->Kpos_D_limit);
     self->i_error_pos = vclampnorm(vadd(self->i_error_pos, vscl(dt, plpos_e)), self->Kpos_I_limit);
 
+    // payload orientation errors
+    // payload quat to R 
+    struct mat33 Rp = quat2rotmat(plquat);
+    // define desired payload Rp_des = eye(3) (i.e., qp_des = [0,0,0,1] (x,y,z,w))
+    struct mat33 Rp_des = quat2rotmat(mkquat(0,0,0,1)); 
```

Comment:
use the desired state from the setpoint instead

Reply:
done